float64[] box_data
